tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -234,6 +240,25 @@ public:
    /// \return A pointer to the newly created RNTupleProcessor.
    static std::unique_ptr<RNTupleProcessor>
    CreateChain(const std::vector<RNTupleOpenSpec> &ntuples, std::unique_ptr<RNTupleModel> model = nullptr);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTuple processor for horizontallly concatenated RNTuples.
+   ///
+   /// \param[in] ntuples A list specifying the names and locations of the ntuples to process. The first ntuple in the
+   /// list will be considered the primary ntuple and drives the processor iteration loop. Subsequent ntuples are
+   /// considered auxiliary, whose entries to be read are determined by the primary ntuple (which does not necessarily
+   /// have to be sequential).
+   /// \param[in] joinFields The names of the fields on which to join, in case the specified ntuples are unaligned.
+   /// These join fields have to be present in every ntuple specified. If an empty list is provided, it is assumed that
+   /// the specified ntuple are fully aligned, and `RNTupleIndex` will not be used.
+   /// \param[in] models A list of models for the ntuples. This list must either contain a model for each ntuple in
+   /// `ntuples` (following the specification order), or be empty. When the list is empty, the default model (i.e.
+   /// containing all fields) will be used for each ntuple.
+   ///
+   /// \return A pointer to the newly created RNTupleProcessor.
+   static std::unique_ptr<RNTupleProcessor> CreateJoin(const std::vector<RNTupleOpenSpec> &ntuples,
+                                                       const std::vector<std::string> &joinFields,
```

Comment:
From the documentation, it's not clear to me if there is one field for each ntuple, or all of them are used to join all ntuples. Either way, is this enough flexibility, or do we need individual sets of join fields for every auxiliary ntuple?

Reply:
I've tried clarifying further in the documentation, let me know if this clears things up. All fields are used to join all ntuples. From the requirements we have now, this seems to be flexible enough, so unless specifically requested I don't think we should change this to sets of join fields.